packages/initiatives/src/follow.ts

```diff
@@ -61,34 +126,67 @@ export function followInitiative(
  * })
  *   .then(response)
  * ```
- * Follow an initiative.
+ * Un-follow an initiative.
  */
 export function unfollowInitiative(
   requestOptions: IFollowInitiativeOptions
 ): Promise<{ success: boolean; username: string }> {
   // we dont call getUser() because the tags are cached and will be mutating
   return request(getUserUrl(requestOptions.authentication), {
     authentication: requestOptions.authentication
-  }).then(user => {
-    // don't update if already following
-    if (!isUserFollowing(user, requestOptions.initiativeId)) {
-      return Promise.reject(`user is not following this initiative.`);
-    }
-
-    const tag = getTag(requestOptions.initiativeId);
-    const tags = JSON.parse(JSON.stringify(user.tags));
-    // https://stackoverflow.com/questions/9792927/javascript-array-search-and-remove-string
-    const index = tags.indexOf(tag);
-    tags.splice(index, 1);
+  })
+    .then(user => {
+      // don't update if not already following
+      if (!isUserFollowing(user, requestOptions.initiativeId)) {
+        return Promise.reject(`user is not following this initiative.`);
+      }
+      // if not already following, pass the user on
+      return user;
+    })
+    .then(user => {
+      const tag = getUserTag(requestOptions.initiativeId);
+      const tags = JSON.parse(JSON.stringify(user.tags));
```

Comment:
I think you could use `cloneObject` here... but since the tags array is primitive types I don't think you'd really need to b/c those are by-val anyhow... I think :)